[cmd/telemetrygen] Flaky test TestRateOfMetrics

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4816347952/jobs/8575876882?pr=20282

```
--- FAIL: TestRateOfMetrics (0.51s)
    worker_test.go:73: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/telemetrygen/internal/logs/worker_test.go:73
        	Error:      	Should be true
        	Test:       	TestRateOfMetrics
        	Messages:   	there should have been more than 6 logs, had 5
2023-04-27T06:03:39.998Z	INFO	logs/logs.go:95	generation of logs isn't being throttled
2023-04-27T06:03:41.095Z	INFO	logs/worker.go:72	logs generated	{"worker": 0, "logs": 84291}
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/cmd/telemetrygen/internal/logs	2.801s
```

--- a/cmd/telemetrygen/internal/logs/worker_test.go
+++ b/cmd/telemetrygen/internal/logs/worker_test.go
@@ -35,7 +35,7 @@ func (m *mockExporter) export(logs plog.Logs) error {
 	return nil
 }
 
-func TestFixedNumberOfMetrics(t *testing.T) {
+func TestFixedNumberOfLogs(t *testing.T) {
 	cfg := &Config{
 		Config: common.Config{
 			WorkerCount: 1,
@@ -55,7 +55,7 @@ func TestFixedNumberOfMetrics(t *testing.T) {
 	require.Len(t, exp.logs, 5)
 }
 
-func TestRateOfMetrics(t *testing.T) {
+func TestRateOfLogs(t *testing.T) {
 	cfg := &Config{
 		Config: common.Config{
 			Rate:          10,
@@ -70,7 +70,7 @@ func TestRateOfMetrics(t *testing.T) {
 
 	// verify
 	// the minimum acceptable number of logs for the rate of 10/sec for half a second
-	assert.True(t, len(exp.logs) >= 6, "there should have been more than 6 logs, had %d", len(exp.logs))
+	assert.True(t, len(exp.logs) >= 5, "there should have been 5 or more logs, had %d", len(exp.logs))
 	// the maximum acceptable number of logs for the rate of 10/sec for half a second
 	assert.True(t, len(exp.logs) <= 20, "there should have been less than 20 logs, had %d", len(exp.logs))
 }

